src/components/modals/ShareModal.vue

```diff
@@ -0,0 +1,111 @@
+<template>
+	<Modal title="Export / Share" @closed="$emit('closed')">
+		<ul class="modal-list">
+			<li v-for="component in components" :key="component">
+				<component :is="component" v-bind="componentProps" />
+			</li>
+		</ul>
+	</Modal>
+</template>
+
+<script>
+import Modal from './Modal.vue';
+import { Job, Service } from '@openeo/js-client';
+import Config from '../../../config';
+
+// Load all sharing components defined the config
+const serviceComponents = (Config.supportedBatchJobSharingServices || []).concat(Config.supportedWebServiceSharingServices || []);
+let components = { Modal };
+for(let component of serviceComponents) {
+	components[component] = () => import(`../share/${component}.vue`);;
+}
+
+export default {
+	name: 'ShareModal',
+	components,
+	props: {
+		context: {
+			type: Object,
+			required: true
+		}
+	},
+	computed: {
+		isJob() {
+			return this.context instanceof Job;
+		},
+		isWebService() {
+			return this.context instanceof Service;
+		},
+		components() {
+			if (this.isJob && Array.isArray(Config.supportedBatchJobSharingServices)) {
+				return Config.supportedBatchJobSharingServices;
+			}
+			else if (this.isWebService && Array.isArray(Config.supportedWebServiceSharingServices)) {
+				return Config.supportedWebServiceSharingServices;
+			}
+			return [];
+		},
+		componentProps() {
```

Comment:
I believe there's currently a mismatch between the props defined here and the mandatory prop `stac` as defined in ShareMixin.js

When i tried to adapt the EDC sharing snippet to this plugin system, i had to change the props to job here:
https://github.com/eurodatacube/openeo-web-editor/commit/e7253f848000672234463fb74e496d925db0abbb

However this would probably break sharing in case of services.

Reply:
Indeed, I wasn't fully finished with the implementation. I'm about to update this to something more useful where you don't need to do a lot of work in the plugins.

Reply:
@totycro I've implemented a more feature-complete version of the functionality, the first version that you had a look at was not really complete yet. Please have another look. It should give various options to customize for your needs, e.g. let them provide their user account or so.

![image](https://user-images.githubusercontent.com/8262166/182179526-8ad245e5-7d50-4117-882e-918dfd2c9ab9.png)

I think your existing component transferred to the new plugin system would look like this:

```
<template>
	<ShareInterface
		id="share-edc" title="EDC" :description="description" :action="upload"
		actionDefaultIcon="https://avatars.githubusercontent.com/u/53830971?s=200&v=4" @stateChanged="state => this.state = state">
	</ShareInterface>
</template>

<script>
import ShareMixin from './ShareMixin';
import ShareInterface from './ShareInterface.vue';

export default {
    name: 'EDCShare',
	components: {
		ShareInterface
	},
	mixins: [
		ShareMixin
	],
	data() {
		return {
			state: 'default'
		};
	},
	computed: {
		description() {
			if (this.state === 'error') {
				return 'Publishing to eurodatacube failed';
			}
			else if (this.state === 'success') {
				return 'Published on eurodatacube';
			}
			else {
				return 'Publish results on eurodatacube (EDC)';
			}
		}
	},
    methods: {
        upload() {
            const formData = new FormData();
            formData.append("my-file.json", JSON.stringify(this.extra));
            return axios.post(
                '/api/upload',
                formData,
            );
        }
    }
}
</script>

<style lang="scss">
#share-edc {
	&.shareable > .entry {
		color: #004170;
	}
}
</style>
```

Reply:
@m-mohr Thanks, this already seems to be pretty close to what we need!

Though currently I'm getting the message `"Sorry, this job has no public URL"` for my jobs, which as far as I understand shouldn't be a problem for sharing the results on EDC since we post the stac item containing the asset links to the backend. Do you think a restricted sharing modal could be shown for such cases where no url is available?

Concerning user customization: I don't really know enough about the context of the editor to be able to say how this should be handled, e.g. if there's a single deployment of the editor or if it's going to be deployed for each customer/user. So it's possible that we'll need more customizations regarding where to post the results to.

Reply:
Yeah, the functionality is meant for sharing so having a public URL seems necessary. Would it be a possibility to add the canonical link to the response? This enables the sharing functionality. Otherwise, it would need more work on the Editor side to make this configurable per "plugin" which I don't have and would expect a contribution from your side.

I'm also not sure about customization as this is closely bound to your project which I'm not part of. But usually you don't need a deployment per end-user, but instead it is deployed per project (if it needs custom code in the Editor, otherwise use editor.openeo.org).

Reply:
Thanks for your comments! I'll discuss internally in my company about how we're going to proceed based on this plugin interface.